airbyte-integrations/connectors/source-tidb/src/main/java/io/airbyte/integrations/source/tidb/TiDBSourceOperations.java

```diff
@@ -0,0 +1,69 @@
+package io.airbyte.integrations.source.tidb;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.mysql.cj.MysqlType;
+import com.mysql.cj.jdbc.result.ResultSetMetaData;
+import com.mysql.cj.result.Field;
+import io.airbyte.db.SourceOperations;
+import io.airbyte.integrations.source.mysql.MySqlSourceOperations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static io.airbyte.db.jdbc.JdbcConstants.*;
+
+public class TiDBSourceOperations extends MySqlSourceOperations implements SourceOperations<ResultSet, MysqlType> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TiDBSourceOperations.class);
+
+  @Override
+  public void setJsonField(ResultSet resultSet, int colIndex, ObjectNode json) throws SQLException {
+    final ResultSetMetaData metaData = (ResultSetMetaData) resultSet.getMetaData();
+    final Field field = metaData.getFields()[colIndex - 1];
+    final String columnName = field.getName();
+    final MysqlType columnType = field.getMysqlType();
+
+    switch (columnType) {
+      case BIT -> {
+        putBinary(json, columnName, resultSet, colIndex);
+      }
```

Comment:
Can you link to TiDB docs showing BIT works only a binary?

Reply:
The formal doc of BIT is https://docs.pingcap.com/tidb/stable/data-type-numeric#bit-type.